src/CMakeLists.txt

```diff
@@ -623,12 +621,20 @@ if(${WITH_RADOSGW})
     expat
     fcgi
   )
-endif(${WITH_RADOSGW})
-if(${WITH_RBD})
+endif(WITH_RADOSGW)
+if(WITH_RBD)
   set(DENCODER_EXTRALIBS
       ${DENCODER_EXTRALIBS}
       rbd_replay_types)
-endif(${WITH_RBD})
+  list(APPEND dencoder_srcs
+      $<TARGET_OBJECTS:krbd_objs>)
+endif(WITH_RBD)
+if(WITH_KVS)
```

Comment:
sorry, i was asking why we needed to guard `$<TARGET_OBJECTS:parse_secret_objs>` with `WITH_KVS`.